vcf-dumper/vcf-dumper-lib/src/main/java/uk/ac/ebi/eva/vcfdump/VariantExporterController.java

```diff
@@ -221,7 +238,25 @@ public void run() {
         VCFHeader header = getOutputVcfHeader();
         VariantContextWriter writer = getWriter();
         writer.writeHeader(header);
+        writeVariants(writer);
+    }
 
+    public void writeHeader() {
+        VCFHeader header = getOutputVcfHeader();
+        VariantContextWriter writer = getWriter();
+        writer.writeHeader(header);
+        writer.close();
+        logger.info("VCF headers exported");
+    }
+
+    public void writeBlock() {
+        VCFHeader header = getOutputVcfHeader();
+        VariantContextWriter writer = getWriter();
+        writer.setVCFHeader(header);
+        writeVariants(writer);
```

Comment:
I see. Can you move the writer.close() outside writeVariants, to make the three variants (write complete file, write just header, write block) more consistent?